pkg/app/api/grpcapi/piped_api.go

```diff
@@ -1112,6 +1112,7 @@ func (a *PipedAPI) CreateDeploymentChain(ctx context.Context, req *pipedservice.
 				},
 			},
 		},
+		Status:    model.ChainBlockStatus_DEPLOYMENT_BLOCK_RUNNING,
```

Comment:
Isn't this PENDING?

Reply:
No, as defined here, this block has a node with defined deployment thus it's counted as RUNNING
https://github.com/pipe-cd/pipe/pull/2866/files#diff-2e2aef3e992fccccb618d2766a559166a34ea41c577b3055516ea48bef6bc399R64

Reply:
I think to be simple (at least for our users when seeing on the web console), the status of the block should be unified with the status of the deployment.
Since the first deployment was created but it is still under `PENDING` status so this block should be at `PENDING` too.
Block should be at `RUNNING` only if one of its deployments is at `RUNNING` status.

How do you think?

Reply:
> Block should be at RUNNING only if any of its deployments is at RUNNING status.

Well, I don't have a strong opinion on this, but block status is different from its node's status is understandable to me (for example, even some node is at SUCCESS status, the block status can be FAILURE by just one node failed), so the problem is in the code (our reading space) or in the viewer (on the web console). 

If for our sake of simplicity (our codebase space), the current block status definition is simpler since in fact, we only care about the start and the finished status of the block, once block initialized count it as PENDING, one some piped submit deploymentRef => count block as RUNNING - and it will be always at the time the deployment is added to block [code at here](https://github.com/pipe-cd/pipe/pull/2866/files#diff-ea408769a64531a39bf46a5a7828335047aea7b3331d4bd3922d587891d6b803R32) we don't need to care about when the deployment status is updated or anything.

If for user's sake of simplicity, I think it depends since if the UI is defined well enough, it will be
- chain - block is displayed on a different page with deployment itself so user can't see both at the same time
- block contains nodes (with nodes' status or deployments' status) and the container is counted as RUNNING while the inside is counted as PENDING means the container is running at the phase to prepare for the inside item to start RUNNING, that's quite simple to understand.

How do you think about that?

Reply:
On the second thought, we create deployments for all applications in the chain even before it's started running, so if the block in the rear of the chain (which has not yet started any of its deployments) and all of that block's deployments are created, it will be marked as RUNNING. That may cause users to misunderstand since that block does not deploy anything, right?

Reply:
Addressed by https://github.com/pipe-cd/pipe/pull/2866/commits/15966954e43f2137e5e25fdfc275b878995e6cb2

Reply:
> we create deployments for all applications in the chain even before it's started running, so if the block in the rear of the chain (which has not yet started any of its deployments) and all of that block's deployments are created, it will be marked as RUNNING. That may cause users to misunderstand since that block does not deploy anything, right?

Yes, that was my point. All deployments in a block will be stayed at `PENDING` after being created to wait for handling by their Pipeds. So the block should be at `PENDING` for more understanding.

Reply:
Sure, addressed it already, so PTAL 😉